### PR TITLE
Refactor stack and mutexset manipulation

### PR DESCRIPTION
**Pull request by xairy**

### CODE REVIEW

File: mm/ktsan/thr.c [hunk 1]
```diff
@@ -221,13 +221,11 @@ void kt_thr_interrupt(kt_thr_t *thr, uptr_t pc, kt_interrupted_t *state)
 	 * BUG_ON(thr->irqs_disabled);
 	 */
 
-	memcpy(&state->stack.pc[0], &thr->stack.pc[0],
```
Comment (dvyukov) [inline mm/ktsan/thr.c]: This was done on a purpose: we don't copy whole stack. Now you copy whole stack. Real stack can be much smaller than capacity. Please revert.

**Comment by dvyukov** [comment]
LGTM if you revert the memcpy

**Comment by xairy** [comment]
I added kt_stack_copy instead. Is that good?

**Comment by dvyukov** [comment]
looks good to me

**Pull request merged**